be1-go/internal/popserver/handler/query.go

```diff
@@ -0,0 +1,286 @@
+package handler
+
+import (
+	"encoding/json"
+	"popstellar/internal/popserver/config"
+	"popstellar/internal/popserver/database"
+	"popstellar/internal/popserver/state"
+	jsonrpc "popstellar/message"
+	"popstellar/message/answer"
+	"popstellar/message/query"
+	"popstellar/message/query/method"
+	"popstellar/network/socket"
+)
+
+func handleQuery(socket socket.Socket, msg []byte) *answer.Error {
+	var queryBase query.Base
+
+	err := json.Unmarshal(msg, &queryBase)
+	if err != nil {
+		errAnswer := answer.NewInvalidMessageFieldError("failed to unmarshal: %v", err).Wrap("handleQuery")
+		socket.SendError(nil, errAnswer)
+		return errAnswer
+	}
+
+	var id *int = nil
+	var errAnswer *answer.Error
+
+	switch queryBase.Method {
+	case query.MethodCatchUp:
+		id, errAnswer = handleCatchUp(socket, msg)
+	case query.MethodGetMessagesById:
+		id, errAnswer = handleGetMessagesByID(socket, msg)
+	case query.MethodGreetServer:
+		id, errAnswer = handleGreetServer(socket, msg)
+	case query.MethodHeartbeat:
+		errAnswer = handleHeartbeat(socket, msg)
+	case query.MethodPublish:
+		id, errAnswer = handlePublish(socket, msg)
+	case query.MethodSubscribe:
+		id, errAnswer = handleSubscribe(socket, msg)
+	case query.MethodUnsubscribe:
+		id, errAnswer = handleUnsubscribe(socket, msg)
+	default:
+		errAnswer = answer.NewInvalidResourceError("unexpected method: '%s'", queryBase.Method)
+	}
+
+	if errAnswer != nil {
+		errAnswer = errAnswer.Wrap("handleQuery")
+		socket.SendError(id, errAnswer)
```

Comment:
this seems wrong - you only need to respond to a JSON-RPC, not a JSON Notification. I believe `Heartbeat` and `GreetServer` are notifications.